### ISSUE TITLE
Optional properties are not generated for objects when there is only one property specified

### ISSUE THREAD

**Issue by Stranger6667** [Zac-HD/hypothesis-jsonschema#25]
Hello!

I found some unexpected behavior in object generation. Given this schema:

```python
SCHEMA = {
    "properties":  {
        "key": {
            "type": "string"
        }
    }, 
    "additionalProperties": False, 
    "type": "object", 
    "required": []
}
```

Only empty dictionaries are generated, however `{"key": ""}` is a valid example as well. Test case:

```python
from hypothesis import given
from hypothesis_jsonschema import from_schema

schema = from_schema(SCHEMA)

@given(query=schema)
def test(query):
    print(query)
```

Test session output:

```
⇒  pytest test_example.py -s --hypothesis-show-statistics
======================== test session starts ========================
platform linux -- Python 3.7.4, pytest-5.1.2, py-1.8.0, pluggy-0.13.0
rootdir: /tmp, inifile: pytest.ini
plugins: hypothesis-4.50.2, mock-1.10.0, env-0.6.2, asyncio-0.10.0, subtests-0.2.1, recording-0.3.5, xdist-1.26.1, forked-1.0.1, cov-2.6.1, aiohttp-0.3.0, schemathesis-0.18.1
collected 1 item                                                    

test_example.py {}
.
======================= Hypothesis Statistics =======================

test_example.py::test:

  - 1 passing examples, 0 failing examples, 0 invalid examples
  - Typical runtimes: < 1ms
  - Fraction of time spent in data generation: ~ 47%
  - Stopped because nothing left to do

========================= 1 passed in 0.01s =========================
```

Setting `additionalProperties: True` changes the generation process and `key` appears in the generated data. Also, adding a second property makes generation work.

`hypothesis-jsonschema` version - 0.9.10, but the behavior is reproducible with older versions as well (tried up to 0.9.7)

My investigation at this point is:

```python
elements = cu.many(
            draw(st.data()).conjecture_data,
            min_size=min_size,
            max_size=max_size,
            average_size=min(min_size + 5, (min_size + max_size) // 2),
        )
```

`average_size` is 0 here (having `min_size=0` and `max_size=1`) which makes `elements.more()` return `False` immediately and nothing will be generated. Should we use `//` here? with ` `/`  values are generated as expected, but I am not sure how this heuristic is calculated 

What do you think?

**Comment by Zac-HD** [comment]
Definitely a bug, and after some investigation in the Hypothesis internals you got the fix exactly right - I should have been using true division here all along!  

Thanks for the fantastic report; I've shipped a fix in 0.9.11 :smile:

**Comment by Stranger6667** [comment]
Thank you! :)
